server/athenian/api/experiments/preloading/entries.py

```diff
@@ -52,6 +57,30 @@ async def _fetch_precomputed_releases(cls,
             releases.reset_index(drop=True, inplace=True)
         return releases
 
+    @classmethod
+    @sentry_span
+    async def fetch_precomputed_release_match_spans(
+            cls,
+            match_groups: Dict[ReleaseMatch, Dict[str, List[str]]],
+            account: int,
+            pdb: databases.Database) -> Dict[str, Dict[str, Tuple[datetime, datetime]]]:
+        """Find out the precomputed time intervals for each release match group of repositories."""
+        cached_df = pdb.cache.dfs[PCID.releases_match_timespan]
+        df = cached_df.df
+        mask = cls._match_groups_to_mask(df, match_groups) & (df["acc_id"] == account)
+        release_match_spans = cached_df.filter(mask)
+        spans = {}
+        for row in release_match_spans.iterrows():
```

Comment:
There is nothing slower than `iterrows()`. Avoid it like leprosy :zombie: 
```suggestion
        for time_from, time_to, release_match in zip(release_match_spans["time_from"].values, release_match_spans["time_to"].values, release_match_spans["release_match"].values):
```

Reply:
HAHAHA got it! Thanks a lot! 🙇